go.mod

```diff
@@ -5,9 +5,11 @@ go 1.22.0
 toolchain go1.23.0
 
 require (
+	github.com/Javex/caddy-fail2ban v0.0.0-20240525135356-cbf9f5a003ce
```

Comment:
I'd rather we didn't pull in dependencies like this.

The long-term plan is to move this plugin into Caddy mainline, so we don't want any dependencies that we cannot trust to be maintained.

Reply:
I have removed the external dependency and implemented the function to be part of this repository.
I would appreciate a review.